nano/core_test/bootstrap.cpp

```diff
@@ -1306,6 +1306,12 @@ TEST (bootstrap_processor, lazy_pruning_missing_block)
 	config.frontiers_confirmation = nano::frontiers_confirmation_mode::disabled;
 	config.enable_voting = false; // Remove after allowing pruned voting
 	nano::node_flags node_flags;
+	// It looks like force confirmed elections remain in the AEC and
+	// they keep publishing blocks, disabling the request loop stops this.
+	// A better fix would be to create a way to add and cement the blocks
+	// without involving an election at all or have a way to delete an
+	// election from the AEC.
+	node_flags.disable_request_loop = true;
```

Comment:
This comment suggests that inserting and confirming blocks directly in the ledger `node->ledger.process` and `node->ledger.confirm` would be a better solution.

Reply:
If we have a way to insert and confirm without creating an election then that might be a cleaner solution, if I understood what is happening well.
I was not aware of `nano::ledger::confirm`.
I will try it later.

Reply:
For reference it was recently added here https://github.com/nanocurrency/nano-node/pull/4519